### PR TITLE
New brand styles integration

### PR DESCRIPTION
**Pull request by octavioamu**
##### Description

**EPIC** - New brand styles integration

This PR:

- Builds on top of #8360
- Copies all stylesheets from ./app/assets/v2/css/\*\*/\*\*/\*.css (and other locations) -> ./app/assets/v2/scss/\*\*/\*\*/\*.scss
- Internalises vue-innersearch.css and quill.\*.css (as .scss)
- Changes all stylesheet paths to sass equivalents (with django-libsass block's) in templates (* ~100 files)
- Removes all stylesheets from ./app/assets/v2/css/\*\*/\*\*/\*.css
- Drops preloads from shared/head.html
- Fixes stylelint path and introduces stylelint-scss (temporarily adding `no-descending-specificity` until we've sorted out nesting in the sass files)
- Fixes stylesheets manually and with `stylelint --fix`
- Drops typography.css from email templates (ignored by premailer) and tests
- Adds collectstatic to travis build to enable tests to pass (with precompilation over sass assets)
- Fixes minor styling bug on quests (global .btn being overridden by page styles)
- Drops stylesheets that don't exist (having checked against live site)
- Adds offline precompilation steps for production
- Adds JS bundling to further reduce requests
- Removes duplicated script and stylesheet inclusions
- Removes jquery-old where appropriate (moving select2 out of the if slim block)
- Adds jsrender to the if slim block

##### Refers/Fixes

Refers: #8363
Refers: #8308
Refers: #8358

### CODE REVIEW

**Review by chibie — APPROVED** [review]

**Comment by thelostone-mc** [comment]
DAMN THIS IS A FANCY PR @gdixon  @octavioamu

File: app/app/templates/shared/tip_dependancies.html [hunk 1]
```diff
@@ -14,12 +14,14 @@
...
+
 <script src="/dynamic/js/tokens_dynamic.js"></script>
 <script src="{% static "v2/js/lib/ipfs-api.js" %}"></script>
+{% compress js file tip_dependancies %}
+<script src="{% static "v2/js/tokens.js" %}"></script>
 <script src="{% static "v2/js/ipfs.js" %}"></script>
 <script src="{% static "v2/js/lib/secrets.min.js" %}"></script>
 <script src="{% static "v2/js/ethereumjs-accounts.js" %}"></script>
 <script src="{% static "onepager/js/send.js" %}"></script>
-<script src="{% static "v2/js/truncate-hash.js" %}"></script>
```
Comment (thelostone-mc) [inline app/app/templates/shared/tip_dependancies.html:25]: is this not needed ?

File: app/retail/templates/admin/index.html [hunk 2]
```diff
@@ -17,9 +17,13 @@
...
 
-{% load i18n static %}
+{% load i18n static compress %}
 
-{% block extrastyle %}{{ block.super }}<link rel="stylesheet" type="text/css" href="{% static "admin/css/dashboard.css" %}">{% endblock %}
+{% block extrastyle %}{{ block.super }}
+    {% compress css file admin_dashboard %}
+        <link rel="stylesheet" type="text/x-scss" href={% static "v2/scss/admin/dashboard.scss" %} />
+    {% endcompress %}
+{% endblock %}
```
Comment (thelostone-mc) [inline app/retail/templates/admin/index.html:26]: re-indent with 2 spaces

**Review by thelostone-mc — APPROVED** [review]
minor comments but other wise I am excited for this PR

File: app/app/templates/shared/tip_dependancies.html (hunk 1 above)
Comment (octavioamu) [reply app/app/templates/shared/tip_dependancies.html:25]: is already loaded on footer_scripts

File: app/retail/templates/admin/index.html (hunk 2 above)
Comment (gdixon) [reply app/retail/templates/admin/index.html:24]: That is done, and I've fixed another ~20 templates too!

**Pull request merged**